clustering/train.py

```diff
@@ -22,46 +25,71 @@ def __init__(self, pd_rules, n_clusters_low=2, n_clusters_high=50, n_clusters_st
         self.n_clusters_optimal = None
         self.kmeans_optimal = None
 
-    def preprocess(self, df, categorical_cols=None, drop_cols=None):
-        df = self.preprocess_dropcols(df, drop_cols=drop_cols)
-        df = self.preprocess_onehot(df, categorical_cols=categorical_cols)
-        df = self.preprocess_fillnulls(df)
-
-        return df
-
-    def preprocess_onehot(self, df, categorical_cols=None):
+        self.categorical_cols = categorical_cols
+        self.drop_cols = drop_cols
+
+    def train_the_cluster(self):
```

Comment:
@TreeinRandomForest please have a look at my refactoring here. I basically want to move all to a single method like [here](https://github.com/ManageIQ/aiops-insights-clustering/pull/21/commits/b8c85c739428e6c758afb992bcdaf4d925e36163#diff-3f41e546893dc64b71aaacad12cad815R23), which just creates `Cluster` and then calls `train_...`

Reply:
@durandom, that looks good - I am also making the following changes:

1. Create a config.py file that contains attributes for
a) Preprocess settings: categorical_cols, drop_cols, index_cols
b) KMeans settings: n_clusters_low, n_clusters_high, n_clusters_stepsize
c) Storage settings: local vs ceph etc. and any relevant attribute
...
This can then be used both during train and inference time without risk of the preprocessing pipelines being different.

2. train.Cluster is a mess right now because some arguments are passed to the constructor as attributes while many of the functions are really stand-alone and require every argument to be passed explicitly. I'll make it more of a "closed" box where the configurations are passed to the constructor like you did with the preprocess function.